// @shopify/shopify-api/dist/cjs/rest/admin/2023-10/usage_charge.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');
var currency = require('./currency.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class UsageCharge extends base.Base {
    static async find({ session, id, recurring_application_charge_id = null, fields = null }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "id": id, "recurring_application_charge_id": recurring_application_charge_id },
            params: { "fields": fields },
        });
        return result.data ? result.data[0] : null;
    }
    static async all({ session, recurring_application_charge_id = null, fields = null, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: { "recurring_application_charge_id": recurring_application_charge_id },
            params: { "fields": fields, ...otherArgs },
        });
        return response;
    }
}
UsageCharge.apiVersion = types.ApiVersion.October23;
UsageCharge.hasOne = {
    "currency": currency.Currency
};
UsageCharge.hasMany = {};
UsageCharge.paths = [
    { "http_method": "get", "operation": "get", "ids": ["recurring_application_charge_id"], "path": "recurring_application_charges/<recurring_application_charge_id>/usage_charges.json" },
    { "http_method": "get", "operation": "get", "ids": ["recurring_application_charge_id", "id"], "path": "recurring_application_charges/<recurring_application_charge_id>/usage_charges/<id>.json" },
    { "http_method": "post", "operation": "post", "ids": ["recurring_application_charge_id"], "path": "recurring_application_charges/<recurring_application_charge_id>/usage_charges.json" }
];
UsageCharge.resourceNames = [
    {
        "singular": "usage_charge",
        "plural": "usage_charges"
    }
];

exports.UsageCharge = UsageCharge;
