// @shopify/shopify-api/dist/cjs/rest/admin/2024-04/abandoned_checkout.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');
var currency = require('./currency.js');
var customer = require('./customer.js');
var discount_code = require('./discount_code.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class AbandonedCheckout extends base.Base {
    static async checkouts({ session, since_id = null, created_at_min = null, created_at_max = null, updated_at_min = null, updated_at_max = null, status = null, limit = null, ...otherArgs }) {
        const response = await this.request({
            http_method: "get",
            operation: "checkouts",
            session: session,
            urlIds: {},
            params: { "since_id": since_id, "created_at_min": created_at_min, "created_at_max": created_at_max, "updated_at_min": updated_at_min, "updated_at_max": updated_at_max, "status": status, "limit": limit, ...otherArgs },
            body: {},
            entity: null,
        });
        return response ? response.body : null;
    }
}
AbandonedCheckout.apiVersion = types.ApiVersion.April24;
AbandonedCheckout.hasOne = {
    "currency": currency.Currency,
    "customer": customer.Customer
};
AbandonedCheckout.hasMany = {
    "discount_codes": discount_code.DiscountCode
};
AbandonedCheckout.paths = [
    { "http_method": "get", "operation": "checkouts", "ids": [], "path": "checkouts.json" },
    { "http_method": "get", "operation": "checkouts", "ids": [], "path": "checkouts.json" }
];
AbandonedCheckout.resourceNames = [
    {
        "singular": "abandoned_checkout",
        "plural": "abandoned_checkouts"
    }
];

exports.AbandonedCheckout = AbandonedCheckout;
