// @shopify/shopify-api/dist/cjs/rest/admin/2024-04/checkout.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');
var discount_code = require('./discount_code.js');
var order = require('./order.js');
var gift_card = require('./gift_card.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class Checkout extends base.Base {
    static async find({ session, token }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "token": token },
            params: {},
        });
        return result.data ? result.data[0] : null;
    }
    static async shipping_rates({ session, token, ...otherArgs }) {
        const response = await this.request({
            http_method: "get",
            operation: "shipping_rates",
            session: session,
            urlIds: { "token": token },
            params: { ...otherArgs },
            body: {},
            entity: null,
        });
        return response ? response.body : null;
    }
    async complete({ body = null, ...otherArgs }) {
        const response = await this.request({
            http_method: "post",
            operation: "complete",
            session: this.session,
            urlIds: { "token": this.token },
            params: { ...otherArgs },
            body: body,
            entity: this,
        });
        return response ? response.body : null;
    }
}
Checkout.apiVersion = types.ApiVersion.April24;
Checkout.hasOne = {
    "discount_code": discount_code.DiscountCode,
    "order": order.Order
};
Checkout.hasMany = {
    "gift_cards": gift_card.GiftCard
};
Checkout.paths = [
    { "http_method": "get", "operation": "get", "ids": ["token"], "path": "checkouts/<token>.json" },
    { "http_method": "get", "operation": "shipping_rates", "ids": ["token"], "path": "checkouts/<token>/shipping_rates.json" },
    { "http_method": "post", "operation": "complete", "ids": ["token"], "path": "checkouts/<token>/complete.json" },
    { "http_method": "post", "operation": "post", "ids": [], "path": "checkouts.json" },
    { "http_method": "put", "operation": "put", "ids": ["token"], "path": "checkouts/<token>.json" }
];
Checkout.primaryKey = "token";
Checkout.resourceNames = [
    {
        "singular": "checkout",
        "plural": "checkouts"
    }
];

exports.Checkout = Checkout;
