// @shopify/shopify-api/dist/cjs/rest/admin/2024-01/price_rule.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class PriceRule extends base.Base {
    static async find({ session, id }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "id": id },
            params: {},
        });
        return result.data ? result.data[0] : null;
    }
    static async delete({ session, id }) {
        const response = await this.request({
            http_method: "delete",
            operation: "delete",
            session: session,
            urlIds: { "id": id },
            params: {},
        });
        return response ? response.body : null;
    }
    static async all({ session, limit = null, since_id = null, created_at_min = null, created_at_max = null, updated_at_min = null, updated_at_max = null, starts_at_min = null, starts_at_max = null, ends_at_min = null, ends_at_max = null, times_used = null, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: {},
            params: { "limit": limit, "since_id": since_id, "created_at_min": created_at_min, "created_at_max": created_at_max, "updated_at_min": updated_at_min, "updated_at_max": updated_at_max, "starts_at_min": starts_at_min, "starts_at_max": starts_at_max, "ends_at_min": ends_at_min, "ends_at_max": ends_at_max, "times_used": times_used, ...otherArgs },
        });
        return response;
    }
    static async count({ session, ...otherArgs }) {
        const response = await this.request({
            http_method: "get",
            operation: "count",
            session: session,
            urlIds: {},
            params: { ...otherArgs },
            body: {},
            entity: null,
        });
        return response ? response.body : null;
    }
}
PriceRule.apiVersion = types.ApiVersion.January24;
PriceRule.hasOne = {};
PriceRule.hasMany = {};
PriceRule.paths = [
    { "http_method": "delete", "operation": "delete", "ids": ["id"], "path": "price_rules/<id>.json" },
    { "http_method": "get", "operation": "count", "ids": [], "path": "price_rules/count.json" },
    { "http_method": "get", "operation": "get", "ids": [], "path": "price_rules.json" },
    { "http_method": "get", "operation": "get", "ids": ["id"], "path": "price_rules/<id>.json" },
    { "http_method": "post", "operation": "post", "ids": [], "path": "price_rules.json" },
    { "http_method": "put", "operation": "put", "ids": ["id"], "path": "price_rules/<id>.json" }
];
PriceRule.resourceNames = [
    {
        "singular": "price_rule",
        "plural": "price_rules"
    }
];

exports.PriceRule = PriceRule;
