// @shopify/shopify-api/dist/esm/rest/admin/2023-01/payment_gateway.mjs
import { Base } from '../../base.mjs';
import { ApiVersion } from '../../../lib/types.mjs';

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class PaymentGateway extends Base {
    static async find({ session, id }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "id": id },
            params: {},
        });
        return result.data ? result.data[0] : null;
    }
    static async delete({ session, id }) {
        const response = await this.request({
            http_method: "delete",
            operation: "delete",
            session: session,
            urlIds: { "id": id },
            params: {},
        });
        return response ? response.body : null;
    }
    static async all({ session, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: {},
            params: { ...otherArgs },
        });
        return response;
    }
}
PaymentGateway.apiVersion = ApiVersion.January23;
PaymentGateway.hasOne = {};
PaymentGateway.hasMany = {};
PaymentGateway.paths = [
    { "http_method": "delete", "operation": "delete", "ids": ["id"], "path": "payment_gateways/<id>.json" },
    { "http_method": "get", "operation": "get", "ids": [], "path": "payment_gateways.json" },
    { "http_method": "get", "operation": "get", "ids": ["id"], "path": "payment_gateways/<id>.json" },
    { "http_method": "post", "operation": "post", "ids": [], "path": "payment_gateways.json" },
    { "http_method": "put", "operation": "put", "ids": ["id"], "path": "payment_gateways/<id>.json" }
];
PaymentGateway.resourceNames = [
    {
        "singular": "payment_gateway",
        "plural": "payment_gateways"
    }
];

export { PaymentGateway };
