// @shopify/shopify-api/dist/esm/rest/admin/2023-04/dispute_evidence.mjs
import { Base } from '../../base.mjs';
import { ApiVersion } from '../../../lib/types.mjs';
import { Fulfillment } from './fulfillment.mjs';

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class DisputeEvidence extends Base {
    static async find({ session, dispute_id }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "dispute_id": dispute_id },
            params: {},
        });
        return result.data ? result.data[0] : null;
    }
}
DisputeEvidence.apiVersion = ApiVersion.April23;
DisputeEvidence.hasOne = {};
DisputeEvidence.hasMany = {
    "fulfillments": Fulfillment
};
DisputeEvidence.paths = [
    { "http_method": "get", "operation": "get", "ids": ["dispute_id"], "path": "shopify_payments/disputes/<dispute_id>/dispute_evidences.json" },
    { "http_method": "put", "operation": "put", "ids": ["dispute_id"], "path": "shopify_payments/disputes/<dispute_id>/dispute_evidences.json" }
];
DisputeEvidence.primaryKey = "dispute_id";
DisputeEvidence.resourceNames = [
    {
        "singular": "dispute_evidence",
        "plural": "dispute_evidences"
    }
];

export { DisputeEvidence };
