// @shopify/shopify-api/dist/esm/rest/admin/2024-04/shop.mjs
import { Base } from '../../base.mjs';
import { ApiVersion } from '../../../lib/types.mjs';

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class Shop extends Base {
    static async current({ session, fields = null, ...otherArgs }) {
        const result = await this.baseFind({
            session: session,
            urlIds: {},
            params: { "fields": fields, ...otherArgs },
        });
        return result.data ? result.data[0] : null;
    }
    static async all({ session, fields = null, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: {},
            params: { "fields": fields, ...otherArgs },
        });
        return response;
    }
}
Shop.apiVersion = ApiVersion.April24;
Shop.hasOne = {};
Shop.hasMany = {};
Shop.paths = [
    { "http_method": "get", "operation": "get", "ids": [], "path": "shop.json" }
];
Shop.resourceNames = [
    {
        "singular": "shop",
        "plural": "shops"
    }
];

export { Shop };
