// @shopify/shopify-app-remix/dist/esm/server/authenticate/webhooks/authenticate.mjs
import { WebhookValidationErrorReason } from '@shopify/shopify-api';
import { adminClientFactory } from '../../clients/admin/factory.mjs';
import '@remix-run/server-runtime';
import { handleClientErrorFactory } from '../admin/helpers/handle-client-error.mjs';

function authenticateWebhookFactory(params) {
    const { api, config, logger } = params;
    return async function authenticate(request) {
        if (request.method !== 'POST') {
            logger.debug('Received a non-POST request for a webhook. Only POST requests are allowed.', { url: request.url, method: request.method });
            throw new Response(undefined, {
                status: 405,
                statusText: 'Method not allowed',
            });
        }
        const rawBody = await request.text();
        const check = await api.webhooks.validate({
            rawBody,
            rawRequest: request,
        });
        if (!check.valid) {
            if (check.reason === WebhookValidationErrorReason.InvalidHmac) {
                logger.debug('Webhook HMAC validation failed', check);
                throw new Response(undefined, {
                    status: 401,
                    statusText: 'Unauthorized',
                });
            }
            else {
                logger.debug('Webhook validation failed', check);
                throw new Response(undefined, { status: 400, statusText: 'Bad Request' });
            }
        }
        const sessionId = api.session.getOfflineId(check.domain);
        const session = await config.sessionStorage.loadSession(sessionId);
        const webhookContext = {
            apiVersion: check.apiVersion,
            shop: check.domain,
            topic: check.topic,
            webhookId: check.webhookId,
            payload: JSON.parse(rawBody),
            subTopic: check.subTopic || undefined,
            session: undefined,
            admin: undefined,
        };
        if (!session) {
            return webhookContext;
        }
        let admin;
        if (config.future.v3_webhookAdminContext) {
            admin = adminClientFactory({
                params,
                session,
                handleClientError: handleClientErrorFactory({ request }),
            });
        }
        else {
            const restClient = new api.clients.Rest({
                session,
                apiVersion: check.apiVersion,
            });
            const graphqlClient = new api.clients.Graphql({
                session,
                apiVersion: check.apiVersion,
            });
            Object.entries(api.rest).forEach(([name, resource]) => {
                Reflect.set(restClient, name, resource);
            });
            admin = {
                rest: restClient,
                graphql: graphqlClient,
            };
        }
        return {
            ...webhookContext,
            session,
            admin: admin,
        };
    };
}

export { authenticateWebhookFactory };
