// @shopify/api-codegen-preset/dist/api-types.js
'use strict';

var fs = require('fs');
var apiConfigs = require('./helpers/api-configs.js');
var preset = require('./preset.js');
var getSchemaData = require('./helpers/get-schema-data.js');

const shopifyApiTypes = ({ apiType, apiVersion, outputDir = ".", documents = ["**/*.{ts,tsx}", "!**/node_modules"], module, }) => {
    const config = apiConfigs.apiConfigs[apiType];
    const { schema, schemaFile } = getSchemaData.getSchemaData(outputDir, apiType, apiVersion);
    const schemaFileExists = fs.existsSync(`${schemaFile}`);
    return {
        ...(schemaFileExists
            ? {}
            : {
                [schemaFile]: {
                    schema,
                    plugins: ["introspection"],
                    config: { minify: true },
                },
            }),
        [`${outputDir}/${config.typesFile}`]: {
            schema: schemaFileExists ? schemaFile : schema,
            plugins: ["typescript"],
        },
        [`${outputDir}/${config.queryTypesFile}`]: {
            schema: schemaFileExists ? schemaFile : schema,
            preset: preset.preset,
            documents,
            presetConfig: { apiType, module },
        },
    };
};

exports.shopifyApiTypes = shopifyApiTypes;
