// @shopify/shopify-app-remix/dist/esm/server/authenticate/admin/billing/check.mjs
import { HttpResponseError } from '@shopify/shopify-api';
import '@remix-run/server-runtime';
import { redirectToAuthPage } from '../helpers/redirect-to-auth-page.mjs';

function checkBillingFactory(params, request, session) {
    return async function checkBilling(options) {
        const { api, logger, config } = params;
        logger.debug('Checking billing plans', { shop: session.shop, ...options });
        try {
            return await api.billing.check({
                session,
                plans: options.plans,
                isTest: options.isTest,
                returnObject: true,
            });
        }
        catch (error) {
            if (error instanceof HttpResponseError && error.response.code === 401) {
                logger.debug('API token was invalid, redirecting to OAuth', {
                    shop: session.shop,
                });
                await config.sessionStorage.deleteSession(session.id);
                throw await redirectToAuthPage(params, request, session.shop);
            }
            else {
                throw error;
            }
        }
    };
}

export { checkBillingFactory };
