// @shopify/shopify-api/dist/cjs/rest/admin/2023-01/inventory_item.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class InventoryItem extends base.Base {
    static async find({ session, id }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "id": id },
            params: {},
        });
        return result.data ? result.data[0] : null;
    }
    static async all({ session, ids = null, limit = null, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: {},
            params: { "ids": ids, "limit": limit, ...otherArgs },
        });
        return response;
    }
}
InventoryItem.apiVersion = types.ApiVersion.January23;
InventoryItem.hasOne = {};
InventoryItem.hasMany = {};
InventoryItem.paths = [
    { "http_method": "get", "operation": "get", "ids": [], "path": "inventory_items.json" },
    { "http_method": "get", "operation": "get", "ids": ["id"], "path": "inventory_items/<id>.json" },
    { "http_method": "put", "operation": "put", "ids": ["id"], "path": "inventory_items/<id>.json" }
];
InventoryItem.resourceNames = [
    {
        "singular": "inventory_item",
        "plural": "inventory_items"
    }
];

exports.InventoryItem = InventoryItem;
