// @shopify/shopify-api/dist/cjs/rest/admin/2023-01/product_resource_feedback.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class ProductResourceFeedback extends base.Base {
    static getJsonBodyName() {
        return "resource_feedback";
    }
    static async all({ session, product_id = null, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: { "product_id": product_id },
            params: { ...otherArgs },
        });
        return response;
    }
}
ProductResourceFeedback.apiVersion = types.ApiVersion.January23;
ProductResourceFeedback.hasOne = {};
ProductResourceFeedback.hasMany = {};
ProductResourceFeedback.paths = [
    { "http_method": "get", "operation": "get", "ids": ["product_id"], "path": "products/<product_id>/resource_feedback.json" },
    { "http_method": "post", "operation": "post", "ids": ["product_id"], "path": "products/<product_id>/resource_feedback.json" }
];
ProductResourceFeedback.resourceNames = [
    {
        "singular": "product_resource_feedback",
        "plural": "product_resource_feedbacks"
    }
];

exports.ProductResourceFeedback = ProductResourceFeedback;
