// @shopify/shopify-api/dist/cjs/rest/admin/2023-04/shop.js
'use strict';

var base = require('../../base.js');
var types = require('../../../lib/types.js');

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class Shop extends base.Base {
    static async current({ session, fields = null, ...otherArgs }) {
        const result = await this.baseFind({
            session: session,
            urlIds: {},
            params: { "fields": fields, ...otherArgs },
        });
        return result.data ? result.data[0] : null;
    }
    static async all({ session, fields = null, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: {},
            params: { "fields": fields, ...otherArgs },
        });
        return response;
    }
}
Shop.apiVersion = types.ApiVersion.April23;
Shop.hasOne = {};
Shop.hasMany = {};
Shop.paths = [
    { "http_method": "get", "operation": "get", "ids": [], "path": "shop.json" }
];
Shop.resourceNames = [
    {
        "singular": "shop",
        "plural": "shops"
    }
];

exports.Shop = Shop;
