// @shopify/shopify-api/dist/esm/rest/admin/2022-10/abandoned_checkout.mjs
import { Base } from '../../base.mjs';
import { ApiVersion } from '../../../lib/types.mjs';
import { Currency } from './currency.mjs';
import { Customer } from './customer.mjs';
import { DiscountCode } from './discount_code.mjs';

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class AbandonedCheckout extends Base {
    static async checkouts({ session, since_id = null, created_at_min = null, created_at_max = null, updated_at_min = null, updated_at_max = null, status = null, limit = null, ...otherArgs }) {
        const response = await this.request({
            http_method: "get",
            operation: "checkouts",
            session: session,
            urlIds: {},
            params: { "since_id": since_id, "created_at_min": created_at_min, "created_at_max": created_at_max, "updated_at_min": updated_at_min, "updated_at_max": updated_at_max, "status": status, "limit": limit, ...otherArgs },
            body: {},
            entity: null,
        });
        return response ? response.body : null;
    }
}
AbandonedCheckout.apiVersion = ApiVersion.October22;
AbandonedCheckout.hasOne = {
    "currency": Currency,
    "customer": Customer
};
AbandonedCheckout.hasMany = {
    "discount_codes": DiscountCode
};
AbandonedCheckout.paths = [
    { "http_method": "get", "operation": "checkouts", "ids": [], "path": "checkouts.json" },
    { "http_method": "get", "operation": "checkouts", "ids": [], "path": "checkouts.json" }
];
AbandonedCheckout.resourceNames = [
    {
        "singular": "abandoned_checkout",
        "plural": "abandoned_checkouts"
    }
];

export { AbandonedCheckout };
