// @shopify/shopify-api/dist/esm/rest/admin/2023-10/carrier_service.mjs
import { Base } from '../../base.mjs';
import { ApiVersion } from '../../../lib/types.mjs';

/***********************************************************************************************************************
* This file is auto-generated. If you have an issue, please create a GitHub issue.                                     *
***********************************************************************************************************************/
class CarrierService extends Base {
    static async find({ session, id }) {
        const result = await this.baseFind({
            session: session,
            requireIds: true,
            urlIds: { "id": id },
            params: {},
        });
        return result.data ? result.data[0] : null;
    }
    static async delete({ session, id }) {
        const response = await this.request({
            http_method: "delete",
            operation: "delete",
            session: session,
            urlIds: { "id": id },
            params: {},
        });
        return response ? response.body : null;
    }
    static async all({ session, ...otherArgs }) {
        const response = await this.baseFind({
            session: session,
            urlIds: {},
            params: { ...otherArgs },
        });
        return response;
    }
}
CarrierService.apiVersion = ApiVersion.October23;
CarrierService.hasOne = {};
CarrierService.hasMany = {};
CarrierService.paths = [
    { "http_method": "delete", "operation": "delete", "ids": ["id"], "path": "carrier_services/<id>.json" },
    { "http_method": "get", "operation": "get", "ids": [], "path": "carrier_services.json" },
    { "http_method": "get", "operation": "get", "ids": ["id"], "path": "carrier_services/<id>.json" },
    { "http_method": "post", "operation": "post", "ids": [], "path": "carrier_services.json" },
    { "http_method": "put", "operation": "put", "ids": ["id"], "path": "carrier_services/<id>.json" }
];
CarrierService.resourceNames = [
    {
        "singular": "carrier_service",
        "plural": "carrier_services"
    }
];

export { CarrierService };
